File: Rust/src/bin/217_contains_duplicate.rs
use std::collections::HashSet;

impl Solution {
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        let mut hs=HashSet::new();
        for el in nums {
            if hs.contains(&el) {
                return true;
            }
            else {
                hs.insert(el);
            }
        }
        return false;
    }
}


struct Solution { }
fn main() {
    println!("{}",Solution::contains_duplicate([1,2,3,1].to_vec())); // true
    println!("{}",Solution::contains_duplicate([1,2,3,4].to_vec())); // false
    println!("{}",Solution::contains_duplicate([1,1,1,3,3,4,3,2,4,2].to_vec())); // true
}

File: Python/217.ContainsDuplicate.py
class Solution:
    def containsDuplicate(self, nums) -> bool:
        s=set()
        for n in nums:
            if n in s: return True
            else: s.add(n)
        return False

        
s=Solution()
print(s.containsDuplicate([1,2,3,1])) # true
print(s.containsDuplicate([1,2,3,4])) # false
print(s.containsDuplicate([1,1,1,3,3,4,3,2,4,2])) # true